urlce7759b8da_Palachsuper_EBA4_tgz-pJ8-json_TestJ8/src/terminal/service/Service.java

```diff
@@ -0,0 +1,193 @@
+import json.*;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.concurrent.TimeUnit;
+
+import terminal.devices.cashe_machine;
+import terminal.devices.printer;
+import terminal.devices.card_reader;
+import terminal.devices.dispencer;
+import terminal.http_server.HttpServer;
+
+
+public class Service {
+    public static String rate = "32";
+    public Service()
+    {
+
+    }
+
+    public static void main() {
+        WriteLog("=====================================================");
+        HttpServer.main("");
+        WriteLog("sdfdsfsd");
+    }
+
+    public static void get_CPU_hash(){
+        try{
+            Runtime.getRuntime().exec("sh /home/terminal/lib/CPU_hash.sh");
+        }
+        catch (Exception e){
+
+        }
+
+    }
+
+    enum MyEnum {
+        print,
+        dispense,
+        card,
+        cashe
+    }
+
+    public static void Parser_comand(String cmd)
+    {
+        int length_cmd = cmd.length();
+        StringBuilder aaa = new StringBuilder();
+        char str;
+        for (int k=11; k<length_cmd-9;k++)
+        {
+           str= cmd.charAt(k);
+           aaa.append(str);
+        }
+        cmd = aaa+"";
+        WritePaymentLog(cmd, cashe_machine.id_payment);
+        String result = cmd;
+         try
+         {
+             result = java.net.URLDecoder.decode(result, "UTF-8");
+             JSONObject json_cmd = new JSONObject(result);
+             String start_cmd = json_cmd.getString("command");
+             try {
+                 MyEnum e = MyEnum.valueOf(start_cmd);
+                 switch (e) {
+                     case print:
+                         printer.main(json_cmd);  // Отправили на в функцию печать JSON
+                         break;
+                     case dispense:
+                         dispencer.main(cmd);
+                         break;
+                     case card:
+                         card_reader.main(cmd);
+                         break;
+                     case cashe:
+                         cashe_machine.main(json_cmd);
+                         break;
+                 }
+             } catch (IllegalArgumentException iae) {
+
+             }
+             String pr_string = json_cmd.getString("params");
+             JSONObject param = new JSONObject(pr_string);
+         }
+         catch (Exception e){
+
+         }
+    }
+
+    public static void WritePaymentLog(String log, String id_payment){
+        StringBuilder sb = new StringBuilder();
+        try{
+            String path_log = "/home/terminal/Log/LogPayment/Payment_"+id_payment+".txt";
+            File Log_path = new File(path_log);
+            if (!Log_path.exists()){
+                try {
+                    Log_path.createNewFile();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 105:112, 151:159.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.